src/NumericalAlgorithms/Interpolation/LinearRegression.hpp

```diff
@@ -0,0 +1,27 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <array>
+
+namespace intrp {
+/*!
+ * \ingroup NumericalAlgorithmsGroup
+ * \brief A linear regression function.
+ *
+ * A wrapper for gsl linear regression.
+ * Fits the data to \f$y = m x + b\f$. The return value
+ * is an array containing \f$(b, m, \delta b, \delta m)\f$, where
```

Comment:
What do you think about returning a struct instead of an array?

Reply:
In the next PR #4137 I use structured bindings to grab and name the correct fields, which returning an array allows me to do. But I am ok with a struct as well (which would change #4137 only by replacing `intercept` with `fit.intercept` etc so it's not so bad).

Reply:
You can structured-binding a struct.  You get the members in the order they are declared in the struct.